components/feature/media/src/main/java/mozilla/components/feature/media/middleware/LastMediaAccessMiddleware.kt

```diff
@@ -4,34 +4,196 @@
 
 package mozilla.components.feature.media.middleware
 
+import androidx.annotation.VisibleForTesting
 import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.ContentAction
 import mozilla.components.browser.state.action.LastAccessAction
 import mozilla.components.browser.state.action.MediaSessionAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.browser.state.selector.privateTabs
 import mozilla.components.browser.state.state.BrowserState
 import mozilla.components.browser.state.state.TabSessionState
 import mozilla.components.concept.engine.mediasession.MediaSession
 import mozilla.components.lib.state.Middleware
 import mozilla.components.lib.state.MiddlewareContext
 
+private const val MEDIA_NOT_PLAYING_TIMESTAMP = 0L
+
 /**
  * [Middleware] that updates [TabSessionState.lastMediaAccess] everytime the user start playing media
  * or navigates to another page in the last tab that was playing media.
  */
 class LastMediaAccessMiddleware : Middleware<BrowserState, BrowserAction> {
-    @Suppress("ComplexCondition")
+    @VisibleForTesting
+    internal val mediaTabsStatus = mutableMapOf<String, MediaTabStatus>()
```

Comment:
Hm so we don't like to introduce state in the middlewares. We want all state to be stored in the store. Can we just add `MediaStatus` to `TabSessionState` instead? Then we also don't need to delete it manually below when the tab is removed.